konduit-serving-cli/src/main/java/ai/konduit/serving/cli/launcher/command/build/extension/ProfileCommand.java

```diff
@@ -0,0 +1,509 @@
+                break;
+            case DELETE:
+                deleteProfile(profileName);
+                break;
+            case VIEW:
+                viewProfile(profileName);
+                break;
+            case CREATE:
+                createProfile();
+                break;
+        }
+    }
+
+    public static void setDefaultProfile(String profileName) {
+        if(!isProfileExists(profileName)) {
+            log.error("No profile with name {} exists.", profileName);
+        } else {
+            try {
+                FileUtils.writeStringToFile(defaultProfileNamePath, profileName, StandardCharsets.UTF_8);
+                log.error("Successfully set '{}' profile as default.", profileName);
+            } catch (IOException e) {
+                log.error("Unable to set default profile", e);
+            }
+        }
+    }
+
+    public static Profile getDefaultProfile() {
+        try {
+            String profileName = FileUtils.readFileToString(defaultProfileNamePath, StandardCharsets.UTF_8);
+            return getProfile(profileName);
+        } catch (IOException e) {
+            log.error("Unable to get default profile", e);
+            return null;
+        }
+    }
+
+    private void createProfile() {
+        if(isProfileExists(profileName)) {
+            out.format("Profile with name %s already exists.%n", profileName);
+        } else {
+            saveProfile(profileName, fillProfileValues(new Profile()));
+        }
+    }
+
+    private void editProfile() {
+        if(isProfileExists(profileName)) {
+            if(profileName.equals("CPU") || profileName.equals("CUDA")) {
+                out.format("Cannot edit pre-set profiles with name 'CPU' or 'CUDA'.%n");
+                System.exit(1);
+            } else {
+                saveProfile(profileName, fillProfileValues(getProfile(profileName)));
+            }
+        } else {
+            out.format("No profile found with the name of %s%n", profileName);
+        }
+    }
+
+    private Profile fillProfileValues(@NonNull Profile profile) {
+        if(cpuArchitecture != null) {
+            profile.cpuArchitecture(cpuArchitecture);
+        }
+
+        if(operatingSystem != null) {
+            profile.operatingSystem(operatingSystem);
+        }
+        if(computeDevice != null) {
+            profile.computeDevice(computeDevice);
+        }
+
+        if(StringUtils.containsIgnoreCase(profile.computeDevice(), "cuda")) {
+            profile.cpuArchitecture("x86");
+        }
+
+        if(serverTypes != null && !serverTypes.isEmpty()) {
+            profile.serverTypes(serverTypes);
+        }
+        if(additionalDependencies != null && !additionalDependencies.isEmpty()) {
+            profile.additionalDependencies(additionalDependencies);
+        }
+
+        if(StringUtils.containsIgnoreCase(profile.computeDevice(), "cuda")) {
+            String cudaVersion = profile.computeDevice().split("_")[1].trim();
```

Comment:
We should validate format. If user passes in "cuda10.0" we'll get an exception on this line

Reply:
They are validated when setting through the profile command arguments using the json setter: https://github.com/KonduitAI/konduit-serving/blob/6b2bdca23ebb45bb0ba6f6888864ce6900fc413f/konduit-serving-cli/src/main/java/ai/konduit/serving/cli/launcher/command/build/extension/model/Profile.java#L99-L105

Also, when setting them through the constructors: https://github.com/KonduitAI/konduit-serving/blob/6b2bdca23ebb45bb0ba6f6888864ce6900fc413f/konduit-serving-cli/src/main/java/ai/konduit/serving/cli/launcher/command/build/extension/model/Profile.java#L91